src/barbequeue/scheduler/classes.py

```diff
@@ -1,95 +1,77 @@
-import time
-from queue import Full, Empty
+import logging
+from queue import Empty, Full
 from threading import Event
 
-from barbequeue.common.classes import BaseCloseableThread
-from barbequeue.messaging.classes import MessageType
+from barbequeue.common.utils import InfiniteLoopThread
+from barbequeue.messaging.classes import Message, MessageType
 
 
 class Scheduler(object):
-
-    def __init__(self, storage_backend, messaging_backend, worker_backend, incoming_mailbox, worker_mailbox):
-        # TODO: Extend the scheduler module to use the messaging backend,
-        # not talk to the worker backend directly
+    def __init__(self, storage_backend, messaging_backend, incoming_mailbox, worker_mailbox):
         self.incoming_mailbox = incoming_mailbox
         self.worker_mailbox = worker_mailbox
-        self.worker_backend = worker_backend
-
-        # HACK, don't talk to the worker queue directly. Won't work if we're using process-based workers.
-        self.worker_queue = worker_backend.jobqueue
 
         self.storage_backend = storage_backend
         self.scheduler_shutdown_event = Event()
 
         self.messaging_backend = messaging_backend
 
-        self.start_scheduler_thread()
+        self.start_scheduler()
+        self.start_worker_message_handler()
 
-    def start_scheduler_thread(self):
-        self.scheduler_thread = SchedulerThread(worker_queue=self.worker_queue,
-                                                messaging_backend=self.messaging_backend,
-                                                storage_backend=self.storage_backend,
-                                                incoming_message_mailbox=self.incoming_mailbox,
-                                                shutdown_event=self.scheduler_shutdown_event, thread_name="SCHEDULER")
-        self.scheduler_thread.setDaemon(True)
+    def start_scheduler(self):
+        self.scheduler_thread = InfiniteLoopThread(func=self.schedule_next_job, thread_name="SCHEDULER",
+                                                   wait_between_runs=0.5)
```

Comment:
Is `0.5` something that should be configurable? Is it a sensitive value? Should we experiment more with it?